packages/react-components/src/shared/src/useControllableState.ts

```diff
@@ -2,21 +2,21 @@ import { isFunction, isUndefined } from "./assertions";
 import { useCallback, useRef } from "react";
 import { useRefState } from "./useRefState";
 
-function validatePrerequisites(controlledValue: any, initialValue: any) {
+function validatePrerequisites<T>(controlledValue: T, initialValue: T) {
     if (!isUndefined(controlledValue) && !isUndefined(initialValue)) {
         throw new Error(
             "useControllableState - A controllable state value can either have a controlled value or an initial value, but not both."
         );
     }
 }
 
-function ensureControlledStateHaveNotChanged(controlledValue: any, isControlled: any) {
+function ensureControlledStateHaveNotChanged<T>(controlledValue: T, isControlled: boolean) {
     if ((isControlled && isUndefined(controlledValue)) || (!isControlled && !isUndefined(controlledValue))) {
         throw new Error("useControllableState - A controllable state value cannot switch between controlled and uncontrolled. Did you inadvertently set a default value (defaultProps) for your controlled prop?");
     }
 }
 
-function useComputeInitialState<T>(controlledValue: T, initialValue: T, defaultValue: T) {
+function useComputeInitialState<T>(controlledValue: T | undefined, initialValue: T | undefined, defaultValue: T | undefined) {
```

Comment:
Should you also accept `null` ? Since the whole PR is about accepting a controlledValue having a `null` value?